src/genai/_utils/general.py

```diff
@@ -152,8 +153,12 @@ def merge_objects(*objs: Optional[Mapping[_KT, _VT]]) -> dict[_KT, _VT]:
     return result
 
 
-def cast_list(input: Union[TInput, list[TInput]]) -> list[TInput]:
-    return input if isinstance(input, list) else [input]
+def is_sequence(value: Any) -> bool:
+    return False if isinstance(value, str) else isinstance(value, Sequence)
+
+
+def cast_list(input: Union[TInput, Sequence[TInput]]) -> Sequence[TInput]:
+    return input if is_sequence(input) else [input]
```

Comment:
Naming does not correspond to the implementation.

`cast_list` should return `list` (not sequence)